### PR TITLE
Jazz JS-adagrams

### PR DESCRIPTION
**Pull request by jazziesf**
# JS Adagrams
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
--- | ---
What patterns were you able to use from your Ruby knowledge to apply to JavaScript? | The code was similar to the previous project but it gave me a chance to refactor in some areas that were a struggle previously. What I take from Ruby is researching new methods from javascript instead of coding right away. 
Did you need to use different strategies to find information online about JavaScript than you would for Ruby? | The MDN seems to have great material on functions. The source for JavaScript seemed more broad than Ruby. 
What was a challenge you were able to overcome on this assignment? | The challenge is the syntax and this. Although I used it in my code it still is a bit confusing. 
What is your favorite thing about learning a new programming language? | Revisiting Adagrams was was exciting. Although Javascript is a struggle the amount of time it took to finish the project was not nearly as challenging as the first time. 
What is your least favorite thing about learning a new programming language? | Feeling defeated at times. I would suggest recommending videos to watch that gives us examples of what javascript can do. Just for the visual learners in the classroom.  
Do you have any recommendations on how we could improve this project for the next cohort? | I thought it was a good idea to revisit a project we did so early in the program.

### CODE REVIEW

File: src/adagrams.js [hunk 1]
```diff
@@ -1,8 +1,95 @@
...
+       let index = Math.floor(Math.random() * letters.length)
+       hand.push(letters[index]);
+       letters.splice(index, 1);
+    }
+    return hand;
+  }, // drawLetters function
+
+  usesAvailableLetters(input, lettersInHand) {
+    let result = true
+    input.split("").forEach(function (letter) {
```
Comment (CheezItMan) [inline src/adagrams.js:15]: An arrow function would go great here!

File: src/adagrams.js [hunk 2]
```diff
@@ -1,8 +1,95 @@
...
+
+  scoreWord(word) {
+    let sum = 0
+
+    if (word.length >= 7) {
+      sum += 8
+    }
+
+    const points = {
+        "AEIOULNRST" : 1,
```
Comment (CheezItMan) [inline src/adagrams.js:35]: Technically you don't need the quotes around the keys here.

**Comment by CheezItMan** [comment]
# JS Adagrams
## What We're Looking For

Feature | Feedback
--- | ---
**General** |
Answered comprehension questions |  Check, What kinds of videos would be helpful?
Small commits with meaningful commit messages |  Only **two** commits!  Gasp!
**Code Requirements** |
`drawLetters` method |  Check
Uses appropriate data structure to store the letter distribution |  Check
All tests for `drawLetters` pass |  Check
`usesAvailableLetters` method |  Check
All tests for `usesAvailableLetters` pass |  Check
`scoreWord` method |  Check
Uses appropriate data structure to store the letter scores |  Check
All tests for `scoreWord` pass |  Check
`highestScoreFrom` method |  Check
Appropriately handles edge cases for tie-breaking logic |  Check
All tests for `highestScoreFrom` pass |  Check
**Overall** |  Well done, you hit all the learning goals for the project.  I would suggest looking up the `map` and `reduce` functions for JavaScript in the MDN.  They could be helpful, and will be in React.
